### ISSUE TITLE
CLI plugin does not with SWC

### ISSUE THREAD

**Issue by jcwillox** [nestjs/swagger#2493]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Currently, the CLI plugin generates an invalid `metadata.ts` file. It uses invalid import paths on Windows, includes syntax errors with the summary field for `@ApiOperation`, and also generates missing imports for enums or classes which are used in the DTO but not exported (which it doesn't have any issues with when using TSC).

* broken paths on windows
![image](https://github.com/nestjs/swagger/assets/39120423/dac4162d-d156-48d7-895a-62acb623fda8)
* Linux paths (correctly relative to the `src` directory)
![image](https://github.com/nestjs/swagger/assets/39120423/f2352283-4af8-46a4-a72d-60b7ddfff371)
* invalid summary field for endpoints
  it's worth noting that removing the `@ApiOperation` tag or disabling the `introspectComments` option avoids this bug.
![image](https://github.com/nestjs/swagger/assets/39120423/0baa430f-56b5-4e7a-9b0e-f1dfbc98f665)
![image](https://github.com/nestjs/swagger/assets/39120423/6ffb8f7c-b835-4f49-b5e0-5044e633234f)
* missing imports for non-exported enums/classes used in exported DTOs, when exported they are correctly referenced and imported in the `t` types array.
![image](https://github.com/nestjs/swagger/assets/39120423/fbaa2fc2-4aba-4f22-9dfb-2528d13a2f9c)

### Minimum reproduction code

https://stackblitz.com/edit/nestjs-typescript-starter-e46jbt?file=src%2Fmetadata.ts

### Steps to reproduce

Note SWC doesn't work on Stackblitz at the moment so you may want to download it locally, the type-checking does work and the error is present on Stackblitz. Obviously, you'd need a Windows machine for the incorrect path bug.

1. create a nestjs project with swc and swagger
2. create a dto and use the a dto
3. annotate an endpoint with the dto, description and summary.
4. run `nest build -b tsc --type-check`

### Expected behavior

The above errors should not occur, `--type-check` should be compatible with a codebase that worked with TSC, if not, we should mention how to make it compatible in the docs.

### Package version

7.0.4

### NestJS version

10.0.3

### Node.js version

v18.13.0

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [X] Linux

**Comment by kamilmysliwiec** [comment]
Could you test with 7.0.5?

**Comment by rubiin** [comment]
Its mostly related to swagger plugin. Tried with the swagger plugin disabled and it runs without any issue

**Comment by kamilmysliwiec** [comment]
Can you provide a minimum reproduction repository?

**Comment by rubiin** [comment]
i will make one right away

**Comment by rubiin** [comment]
https://stackblitz.com/edit/nestjs-typescript-starter-8dx4lo here is the reproduction. 

`nest start` works.
`nest start -b swc --type-check` does not

**Comment by kamilmysliwiec** [comment]
@rubiin this repo does not use 7.0.5

**Comment by rubiin** [comment]
i will update and ping you back

**Comment by jcwillox** [comment]
Tested with 7.0.6, the `summary` bug is fixed 👍, though the other issues are all still present. I updated the Stackblitz repo with the updates.

@rubiin just a heads up swc won't actually work on web containers (without using `@swc/wasm`), so the "bindings" errors are all expected. I actually noticed the same issue and its also present in my reproduction, just delete the `metadata.ts` file and run either `pnpm build` or `pnpm start` both will report no type errors, however, `pnpm start:dev` (i.e. watch mode) will report errors. This is only on the first run, it will still generate a `metadata.ts` file and fail on the second, so this will definitely be an issue for CI as you likely won't notice it during development.

**Comment by rubiin** [comment]
@kamilmysliwiec just tried . The issue still persists when you do `nest start -b swc --type-check`

**Comment by kamilmysliwiec** [comment]
@kamilmysliwiec what issue specifically?

@jcwillox non-exported enums/classes will be now ignored & excluded from the auto-generated metadata file. To see warnings/errors, one can set the `debug` plugin option to `true` in `nest-cli.json`. Fixed in v7.0.7

![image](https://github.com/nestjs/swagger/assets/23244943/b92ce659-580b-43bb-91d7-09b44aebedc4)
![image](https://github.com/nestjs/swagger/assets/23244943/3902793e-a849-48e5-8aea-fe43cf99c7d9)

**Comment by rubiin** [comment]
I have updated my repro.
`pnpm start:dev` works flawlessly but `pnpm start:swc` aborts with a problem and a metadata is generated

**Comment by jcwillox** [comment]
@kamilmysliwiec awesome thanks for that quick turn around the export stuff all looks to be working well now, the only outstanding issue I believe is the "broken paths on windows" which still prevents the type checker from working.

**Comment by kamilmysliwiec** [comment]
Would you mind checking if paths are still broken in v7.0.10 @jcwillox? 🙏

**Comment by jcwillox** [comment]
Yep, the paths bug is fixed, they are now correctly relative like on Linux. However, I did just notice that the `metadata.ts` doesn't appear to get used, the file is valid and includes all the metadata that it should but the swagger docs don't actually appear to use any of it. Is there something special we have to do to make it use the metadata file?

**Comment by kamilmysliwiec** [comment]
@jcwillox https://docs.nestjs.com/openapi/cli-plugin#swc-builder

**Comment by jcwillox** [comment]
Ahh perfect thanks, totally missed that section. I was actually able to get it to work, however, the docs may trip up a lot of people as I needed to use `await SwaggerModule.loadPluginMetadata(metadata);` to get it to work.

![image](https://github.com/nestjs/swagger/assets/39120423/f087d2ec-8757-4676-b368-243adb092bcd)

One last thing I wanted to check, which I mentioned briefly before but the build command does not wait for the `metadata.ts` file to be generated and will fail unless one exists, I think that also means the `metadata.ts` file it uses could be out of date and will break CI's unless it's checked into version control. What's the intended behaviour here?
![image](https://github.com/nestjs/swagger/assets/39120423/012c4dc7-46c9-4283-a17a-30e52f6797a1)
The second time you run the command it succeeds as expected.

**Comment by kamilmysliwiec** [comment]
Just updated the docs to use `await`, great catch!

> One last thing I wanted to check, which I mentioned briefly before but the build command does not wait for the metadata.ts file to be generated and will fail unless one exists

Build command doesn't wait to run SWC only in the `--watch` mode which shouldn't be used in CIs anyway. It does perform type-checking earlier though, as type-checking is actually required for us to produce the metadata file.

>  I think that also means the metadata.ts file it uses could be out of date and will break CI's unless it's checked into version control.

As long as this file exists, it can be safely out-of-the-date as it will be updated in CI anyway before the application is launched.

**Comment by SpeedySH** [comment]
I don't know, maybe i should create a new bug request, but, this bug relevant to this topic.

The thing in PartialType. It doesn't work with SWC compiler. Like:
`export class UpdateEquipmentDTO extends PartialType(CreateEquipmentDTO) {}`

Made output like that:
_metadata.ts_
`
 models: [
        [
          import("./equipments/DTO/create-equipment.dto"),
          {
            CreateSectionDTO: {
              title: { required: true, type: () => String, minLength: 1 },
              imageURL: { required: false, type: () => String },
            },
            CreateEquipmentDTO: {
              title: { required: true, type: () => String, minLength: 1 },
              shortDescription: {
                required: false,
                type: () => String,
                maxLength: 300,
              },
              description: {
                required: false,
                type: () => String,
                maxLength: 5000,
              },
              price: { required: true, type: () => Number },
              imageURLs: { required: true, type: () => [String] },
              previewImageURL: { required: false, type: () => String },
              sectionId: { required: true, type: () => Number },
            },
          },
        ],
        [
          import("./equipments/DTO/update-equipment.dto"),
          { UpdateEquipmentDTO: {} },
        ],
        ...
`



DISCORD TOPIC: [https://discord.com/channels/520622812742811698/1126116119783161897](url)

**Comment by mhadi512** [comment]
@SpeedySH I'm facing the same issue, it seems that **PartialType**, **PickType** and **OmitType** is not working while using the SWC compiler.
I will use the default tsc compiler till this issue is fixed.
cc: @kamilmysliwiec

**Comment by kamilmysliwiec** [comment]
@SpeedySH @mhadi512 please, create a separate issue with a minimum reproduction repository.
